.github/workflows/deploy-staging.yml

```diff
@@ -48,3 +48,41 @@ jobs:
       RS_STAGING_BUGSNAG_API_KEY: ${{ secrets.RS_STAGING_BUGSNAG_API_KEY }}
       SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
       SLACK_RELEASE_CHANNEL_ID: ${{ secrets.SLACK_RELEASE_CHANNEL_ID_NON_PROD }}
+
+  extract-version:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout source branch
+        uses: actions/checkout@v4
+
+      - name: Determine the SDK version
+        id: determine_version
+        run: |
+          # Default new_version to empty
```

Comment:
Similar to the above comment, there is a neat way to extract the version easily. Let's discuss this offline.
For reference, see `deploy.yml` source code checkout steps. We just need to checkout the appropriate commit and read `package.json`. (ignore the fact that `use_pr_head_sha` input is not used for now)

Reply:
In effect, this version calculation can be moved into the sanity workflow.

Reply:
Addressed